project/npda/views/patient.py

```diff
@@ -137,17 +143,43 @@ def get_queryset(self):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
-        context["pz_code"] = self.request.session.get("pz_code")
-        context["selected_audit_year"] = self.request.session.get(
-            "selected_audit_year", "None"
-        )
+        pz_code = self.request.session.get("pz_code")
+        selected_audit_year = self.request.session.get("selected_audit_year")
+
+        pdu = PaediatricDiabetesUnit.objects.get(pz_code=pz_code) if pz_code else None
+        context["pdu"] = pdu
+
+        submission = None
+        submission_error_count = 0
+
+        # TODO MRB: this should probably be a method on the Submission model?
```

Comment:
As above, the first part of this should be a custom manager method. #332 would avoid an additional `json.loads`